### PR TITLE
Stabilizes unit tests

### LINKED ISSUE #54: Tests fail on master
I noticed this when writing #53 and decided to make a separate issue for it. I'll put up a fix soon.

### PR DESCRIPTION
**Pull request by kengorab**
- Fixes #54
  - Replaces jest with mocha, since I was having issues with jest's auto-mocking
    - I feel that mocha is becoming the standard js unit test runner anyway, and was easy to set up with minimal configuration
  - Uses enzyme for shallow/deep rendering instead of react-addons-testutils

### CODE REVIEW

File: test/HTMLView-test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react'
```
Comment (kengorab) [inline test/HTMLView-test.js]: This test captures all of the previous tests, but uses [enzyme](https://github.com/airbnb/enzyme), an awesome testing library that provides a great API on top of shallow rendering.

File: test/support/setupTestFramework.js [hunk 2]
```diff
@@ -1,3 +1,25 @@
+require('babel-register')({
+  presets: ['es2015', 'es2016', 'react-native'],
+  ignore: /node_modules\/(?!react-native)/,
+})
+require('react-native-mock/mock')
```
Comment (kengorab) [inline test/support/setupTestFramework.js]: Requiring this module has the side-effect of overwriting the `require` function's cache to return a mocked-out version of react-native whenever it's imported.

File: test/support/setupTestFramework.js [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
 
-window.shallowRender = require.requireActual('./shallowRender')
-jest.setMock('react-native', require.requireActual('../../__mocks__/react-native'))
+const jsdom = require('jsdom')
```
Comment (kengorab) [inline test/support/setupTestFramework.js]: Enzyme needs a "dom" context to render into, so we create a small jsdom and expose it as globals.

**Comment by jsdf** [comment]
Thanks!

**Pull request merged**